crypto/keys/mintkey/mintkey.go

```diff
@@ -42,36 +44,55 @@ var BcryptSecurityParameter = 12
 
 // Armor the InfoBytes
 func ArmorInfoBytes(bz []byte) string {
-	return armorBytes(bz, blockTypeKeyInfo)
+	header := map[string]string{
+		"type":    "Info",
+		"version": "0.0.0",
+	}
+	return armor.EncodeArmor(blockTypeKeyInfo, header, bz)
 }
 
 // Armor the PubKeyBytes
-func ArmorPubKeyBytes(bz []byte) string {
-	return armorBytes(bz, blockTypePubKey)
-}
-
-func armorBytes(bz []byte, blockType string) string {
+func ArmorPubKeyBytes(bz []byte, algo string) string {
 	header := map[string]string{
-		"type":    "Info",
-		"version": "0.0.0",
+		"version": "0.0.1",
 	}
-	return armor.EncodeArmor(blockType, header, bz)
+	if algo != "" {
+		header["type"] = algo
+	}
+	return armor.EncodeArmor(blockTypePubKey, header, bz)
 }
 
 //-----------------------------------------------------------------
 // remove armor
 
 // Unarmor the InfoBytes
 func UnarmorInfoBytes(armorStr string) (bz []byte, err error) {
-	return unarmorBytes(armorStr, blockTypeKeyInfo)
+	bz, header, err := unarmorBytes(armorStr, blockTypeKeyInfo)
+	if header["version"] != "0.0.0" {
+		err = fmt.Errorf("unrecognized version: %v", header["version"])
+		return
+	}
+	return
 }
 
 // Unarmor the PubKeyBytes
-func UnarmorPubKeyBytes(armorStr string) (bz []byte, err error) {
-	return unarmorBytes(armorStr, blockTypePubKey)
+func UnarmorPubKeyBytes(armorStr string) (bz []byte, algo string, err error) {
+	bz, header, err := unarmorBytes(armorStr, blockTypePubKey)
```

Comment:
ditto, error check should be placed before the `switch` statement. Also, remove named return variables

Reply:
I feel without the named variable for algo, it might be unclear what that return value is